harmonica/equivalent_layer/utils.py

```diff
@@ -59,6 +59,37 @@ def predict_numba(
             )
 
 
+# Define parallelized version of the utils functions
+if hasattr(predict_numba, "py_func"):
+    predict_numba_parallel = jit(nopython=True, parallel=True)(predict_numba.py_func)
+else:
+    predict_numba_parallel = jit(nopython=True, parallel=True)(predict_numba)
+if hasattr(jacobian_numba, "py_func"):
+    jacobian_numba_parallel = jit(nopython=True, parallel=True)(jacobian_numba.py_func)
+else:
+    jacobian_numba_parallel = jit(nopython=True, parallel=True)(jacobian_numba)
```

Comment:
What about not putting the decorators on the functions and doing:

```python
predict_numba = numba.jit(nopython=True)(predict)
predict_numba_parallel = numba.jit(nopython=True, parallel=True)(predict)
```

This way, there is not `if` block or the cryptic inspection of attributes (which numba might decide to change).

Reply:
You're right about this. The whole `if hasattr(...)` workaround was intended to solve the design error of keeping the `jit` decorator on the functions. Not my best week haha!